packages/obj-logger/src/obj-logger.ts

```diff
@@ -0,0 +1,172 @@
+import { DataStream, StringStream } from "scramjet";
+import { PassThrough, Writable } from "stream";
+import { IObjectLogger, LogEntry, LogLevel } from "@scramjet/types";
+import { getName } from "./utils/get-name";
+
+export class ObjLogger implements IObjectLogger {
+    /**
+     * @type {PassThrough} Stream used to write logs.
+     */
+    outputLogStream = new PassThrough({ objectMode: true });
+
+    /**
+     * Input log stream in object mode.
+     */
+    inputLogStream = new PassThrough({ objectMode: true });
+
+    /**
+     * Input log stream in string mode.
+     */
+    inputStringifiedLogStream = new PassThrough({ objectMode: false });
+
+    /**
+     * Output stream in object mode.
+     */
+    output = new DataStream({ objectMode: true });
+
+    /**
+     * Name used to indicate the source of the log.
+     */
+    name: string;
+
+    /**
+     * Default log object.
+     */
+    baseLog: LogEntry;
+
+    /**
+     * Log level.
+     */
+    logLevel: LogLevel;
+
+    /**
+     * Additional output streams.
+     */
+    outputs: Writable[] = [];
+
+    /**
+     * Logging levels chierarchy.
+     */
+    static levels: LogLevel[] = ["FATAL", "ERROR", "WARN", "INFO", "DEBUG", "TRACE"];
+
+    /**
+     *
+     * @param {any} reference Used to obtain a name for the logger.
+     * @param {LogEntry} baseLog Default log object.
+     * @param {LogLevel} logLevel Log level.
+     */
+    constructor(reference: any, baseLog: LogEntry = {}, logLevel: LogLevel = "TRACE") {
+        this.name = getName(reference);
+
+        this.baseLog = baseLog;
+        this.logLevel = logLevel;
+
+        StringStream
+            .from(this.inputStringifiedLogStream)
+            .JSONParse(true)
+            .pipe(this.inputLogStream);
```

Comment:
Do we need to parse this every time? We could just mux our log into the others. It's ok to leave it like this, but maybe a tech debt entry would be in order?

Reply:
In the present solution, we parse incoming logs to retrieve some of the data from them and complete the data of the logger they come to.

inputStringifiedLogStream is used only when stringified data are sent, for example via http (runner -> instance-adapter)

Reply:
Ok, my comment is not applicable in that case.